### ISSUE TITLE
[Core] Unable to run worker with virtual environment without installing dashboard

### ISSUE THREAD

**Issue by SebastianMorawiec** [ray-project/ray#45410]
### What happened + What you expected to happen

Ive tried to minimize the dependencies inherited from Ray.
I am trying to run Ray Remote with new virtualenv (adding new dependencies).

If I am only installing `pip install ray==2.9.0` I get an error that there is no such package as virtualenv from worker.
I need to install `pip install ray[default]==2.9.0` to make it work, which brings 23 more dependencies bunch of them related to ray-dashboard which I am not interested it.

I can add `virtualenv` to my own dependencies, but I'd have to maintain dependency that actually Ray uses.

TLDR:
Please add your `virtualenv` to default dependencies. 

### Versions / Dependencies

Ray 2.9.0 (same goes for 2.22 tho)

### Reproduction script

n/a

### Issue Severity

High: It blocks me from completing my task.

**Comment by rynewang** [comment]
Do you mean `runtime environment` from Ray? This feature requires `ray[default]`, because we need those dependencies to set up things. If you don't want runtime env, you can create an image and install deps in it, thus avoiding runtime env.

**Comment by SebastianMorawiec** [comment]
I want to be able to run Ray Remotes with runtime environment, without installing Ray Dashboard.
Ray[default] installs (AFAIR) 34 more dependencies than just Ray with no extra.
To run Ray Remote with runtime environments, I need to install Ray and separately install virtualenv dependnecy.
Would be much better IMHO if virutalenv just became standard dependnency without any extras

**Comment by cszhu** [comment]
Hello! This P3 issue has seen no activity in at least a year. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
